server/src/main/java/org/eclipse/hono/server/HonoServer.java

```diff
@@ -109,25 +114,124 @@ public void start(final Future<Void> startupHandler) {
 
         checkStandardEndpointsAreRegistered();
 
-        Future<Void> endpointsTracker = Future.future();
-        startEndpoints(endpointsTracker);
-        endpointsTracker.compose(s -> {
-            final ProtonServerOptions options = createServerOptions();
-            server = ProtonServer.create(vertx, options)
-                    .saslAuthenticatorFactory(saslAuthenticatorFactory)
-                    .connectHandler(this::handleRemoteConnectionOpen)
-                    .listen(honoConfig.getPort(), honoConfig.getBindAddress(), bindAttempt -> {
-                        if (bindAttempt.succeeded()) {
-                            LOG.info("HonoServer running at [{}:{}]", getBindAddress(), getPort());
-                            startupHandler.complete();
-                        } else {
-                            LOG.error("cannot start up HonoServer", bindAttempt.cause());
-                            startupHandler.fail(bindAttempt.cause());
-                        }
-                    });
+        Future<Void> portConfigurationTracker = Future.future();
+        doPortConfigurations(portConfigurationTracker);
+
+        portConfigurationTracker.compose(s -> {
+            Future<Void> endpointsTracker = Future.future();
+            startEndpoints(endpointsTracker);
+            return endpointsTracker;
+        }).compose(s -> {
+            // secure server
+            final Future<Void> startSecurePortTracker = Future.future();
```

Comment:
I think we should refactor this block into a separate method returning the Future. WDYT?